streams/src/main/java/org/apache/kafka/streams/state/internals/RocksDBTimestampedStore.java

```diff
@@ -70,20 +70,7 @@ void openRocksDB(final DBOptions dbOptions,
 
         try {
             db = RocksDB.open(dbOptions, dbDir.getAbsolutePath(), columnFamilyDescriptors, columnFamilies);
-
-            final ColumnFamilyHandle noTimestampColumnFamily = columnFamilies.get(0);
-
-            final RocksIterator noTimestampsIter = db.newIterator(noTimestampColumnFamily);
-            noTimestampsIter.seekToFirst();
-            if (noTimestampsIter.isValid()) {
-                log.info("Opening store {} in upgrade mode", name);
-                dbAccessor = new DualColumnFamilyAccessor(noTimestampColumnFamily, columnFamilies.get(1));
-            } else {
-                log.info("Opening store {} in regular mode", name);
-                dbAccessor = new SingleColumnFamilyAccessor(columnFamilies.get(1));
-                noTimestampColumnFamily.close();
-            }
-            noTimestampsIter.close();
+            setDbAccessor(columnFamilies.get(0), columnFamilies.get(1));
         } catch (final RocksDBException e) {
             if ("Column family not found: : keyValueWithTimestamp".equals(e.getMessage())) {
                 try {
```

Comment:
Maybe add an `INFO` or `WARN` statement here to help with tracing control flow.

Reply:
Not sure -- it's an internal implementation detail that we create a second CF -- if not a WARN for sure, because it's expected to happen. We could de a DEBUG if we want.

Reply:
good point. I'd say just leave it as is.